exe/node/Main.hs

```diff
@@ -52,7 +54,19 @@ main = do
         info (helper <*> Node.nodeOptionsParser cfgPaths)
              (progDesc "Oscoin Node")
 
-    keys         <- runReaderT readKeyPair (Just $ keysDir optPaths)
+    keys         <-
+        flip runReaderT (Just (keysDir optPaths)) $
+            catchJust (\case
+                            e@KeyNotFound{} -> Just e
+                            _               -> Nothing)
+                      readKeyPair
+                      (\e ->
+                            if optCreateKeysIfMissing then do
+                                kp <- liftIO Crypto.generateKeyPair
+                                writeKeyPair kp
+                                pure kp
+                            else
+                                throwM e)
```

Comment:
Would this code clean up with `catches`? 🤔 

```haskell
flip runReaderT (Just (keysDir optPaths)) $
  catches [ Handler (\KeyNotFound{} -> ... )]
```

?

Reply:
marginally